File: oauth2/src/main/java/chatox/oauth2/domain/GlobalBan.java
package chatox.oauth2.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GlobalBan {
    @Id
    private String id;

    @ManyToOne
    @JoinColumn
    private Account bannedAccount;

    private boolean permanent;
    private ZonedDateTime expiresAt;
    private boolean canceled;
    private ZonedDateTime canceledAt;
    private ZonedDateTime updatedAt;

    private String bannedUserId;
}

File: events/src/common/types/GlobalBan.ts
import {User} from "./User";

export interface GlobalBan {
    id: string,
    expiresAt: string,
    createdAt: string,
    createdBy: User,
    bannedUser: User,
    canceled: boolean,
    canceledAt?: string,
    canceledBy?: User,
    permanent: boolean,
    reason: string,
    comment?: string,
    updatedAt?: string,
    updatedBy?: User
}
